internal/pkg/cli/local_run.go

```diff
@@ -406,6 +443,141 @@ func (o *localRunOpts) runContainers(ctx context.Context, imageInfoList []clidep
 	return nil
 }
 
+type envVarValue struct {
+	Value  string
+	Secret bool
+}
+
+// getEnvVars uses env overrides passed by flags and environment variables/secrets
+// specified in the Task Definition to return a set of environment varibles for each
+// continer defined in the TaskDefinition. The returned map is a map of container names,
+// each of which contains a mapping of key->envVarValue, which defines if the variable is a secret or not.
+func (o *localRunOpts) getEnvVars(ctx context.Context, taskDef *awsecs.TaskDefinition) (map[string]map[string]envVarValue, error) {
+	envVars := make(map[string]map[string]envVarValue)
+	for _, ctr := range taskDef.ContainerDefinitions {
+		envVars[aws.StringValue(ctr.Name)] = make(map[string]envVarValue)
+	}
+
+	if err := o.fillEnvOverrides(envVars); err != nil {
```

Comment:
Shall we fill env var and then apply override?

Reply:
good idea!